compiler/rustc_span/src/tests.rs

```diff
@@ -2,18 +2,21 @@ use super::*;
 
 #[test]
 fn test_lookup_line() {
-    let lines = &[BytePos(3), BytePos(17), BytePos(28)];
+    let source = "abcdefghijklm\nabcdefghij\n...".to_owned();
+    let sf =
+        SourceFile::new(FileName::Anon(0), source, BytePos(3), SourceFileHashAlgorithm::Sha256);
+    assert_eq!(sf.lines.as_slice(), &[BytePos(3), BytePos(17), BytePos(28)]);
 
-    assert_eq!(lookup_line(lines, BytePos(0)), -1);
-    assert_eq!(lookup_line(lines, BytePos(3)), 0);
-    assert_eq!(lookup_line(lines, BytePos(4)), 0);
+    assert_eq!(sf.lookup_line(BytePos(0)), None);
+    assert_eq!(sf.lookup_line(BytePos(3)), Some(0));
+    assert_eq!(sf.lookup_line(BytePos(4)), Some(0));
```

Comment:
This doesn't really seem like an improvement ... could you move `lookup_lint` to the test file so it doesn't cause all this churn?

Reply:
I see a few things that are beneficial in this implementation, it uses `None` to express failed lookup instead of `-1`, it doesn't requires casting result from `isize` back into `usize` before using, and it tests actually public API instead of a thin internal wrapper around binary search (although, value of this unit test seems low regardless).

`lookup_line` function no longer exists, so there is no reason in moving and keeping it here.